providers/subclassing_an_existing_provider.md

```diff
@@ -0,0 +1,188 @@
+
+In the case where the existing provider API is already a part of ManageIQ it is possible to "subclass" the existing provider plugin and only implement the differences or possibly just provide a more specific name and logo that users will recognize.
+
+For our example here we are going to add an on-premise version of our existing AwesomeCloud from [Creating a new provider](writing_a_new_provider.md), called AwesomePrivateCloud.  This is going to have an OpenStack compatible API but we are going to make some minor changes to the available options, taking advantage of the fact that we have a specific provider plugin.
+
+### Creating the new plugin
+
+First step is to create a new empty plugin:
+```bash
+$ bundle exec rails generate manageiq:provider ManageIQ::Providers::AwesomePrivateCloud --no-scaffolding --vcr
+** ManageIQ master, codename: Oparin
+      create  
+         run  git init /home/grare/adam/src/manageiq/manageiq/plugins/manageiq-providers-awesome_private_cloud from "."
+
+Initialized empty Git repository in /home/grare/adam/src/manageiq/manageiq/plugins/manageiq-providers-awesome_private_cloud/.git/
+      create  manageiq-providers-awesome_private_cloud.gemspec
+      create  .codeclimate.yml
+      create  .gitignore
+      create  .rspec
+      create  .rspec_ci
+      create  .rubocop.yml
+      create  .rubocop_cc.yml
+      create  .rubocop_local.yml
+      create  .whitesource
+      create  .yamllint
+      create  Gemfile
+      create  LICENSE.txt
+      create  Rakefile
+      create  README.md
+      create  bin/ci/after_script
+      create  bin/rails
+      create  bin/setup
+      create  bin/update
+      create  bundler.d
+      create  bundler.d/.keep
+      create  config/secrets.defaults.yml
+      create  config/settings.yml
+      create  lib/manageiq-providers-awesome_private_cloud.rb
+      create  lib/manageiq/providers/awesome_private_cloud/engine.rb
+      create  lib/manageiq/providers/awesome_private_cloud/version.rb
+      create  lib/tasks/README.md
+      create  lib/tasks_private/spec.rake
+      create  locale
+      create  locale/.keep
+      create  spec/factories
+      create  spec/support
+      create  spec/spec_helper.rb
+      insert  /home/grare/adam/src/manageiq/manageiq/Gemfile
+      create  spec/models/manageiq/providers/awesome_private_cloud
+        gsub  lib/tasks_private/spec.rake
+      insert  .yamllint
+      append  spec/spec_helper.rb
+
+```
+
+Follow the steps from [Creating a new provider](writing_a_new_provider.md) for setting up your new plugin for local development by adding to your `bundler.d/override.rb` and symlinking your `spec/manageiq` directory.
+
+You'll notice however that we chose "--no-scaffolding" option, this is going to allow us to define the classes that we need as subclasses of the OpenStack provider.
```

Comment:
This makes me wonder if we should have a `--subclasses ManageIQ::Providers::AwesomeCloud`, which, if --scaffolding is still set, will automatically scaffold the subclasses

Reply:
Yes that is the plan but until then wanted to document how to currently do it

Reply:
Oh yeah definitely